vmdb/app/controllers/miq_ae_customization_controller/dialogs.rb

```diff
@@ -194,6 +194,7 @@ def dialog_edit
     when 'cancel'
       @edit = session[:edit] = nil # clean out the saved info
       self.x_active_tree = :dialogs_tree
+      @record = identify_record(params[:id], Dialog) if params[:id]
```

Comment:
@GregP instead of setting @record, you can try using session[:edit][:rec_id], if that's not nil show edit cancelled message otherwise show Add cancelled flash message. Then you can move @edit = session[:edit] = nil  to after if/else block. LEt me know if you have questions